src/tools/scripts/deploy.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# exit when any command fails instead of trying to continue on
+set -e
+
+# Check branch is clean first
+if [ -n "$(git status --porcelain)" ]; then 
+  read -r -n 1 -p "Your branch is not clean. Do you still want to deploy? [y/N]: " DEPLOY
+  if [ "$DEPLOY" != "Y" ] && [ "$DEPLOY" != "y" ]; then
+      echo
+      echo "Cancelling deploy"
+      exit 0
+  fi
```

Comment:
We can DRY this block of the code using a function that takes the prompt text as an argument because this logic is being repeated below a few times.

Reply:
Good idea. Done.